crates/red_knot/src/args.rs

```diff
@@ -247,3 +251,17 @@ impl From<OutputFormat> for ruff_db::diagnostic::DiagnosticFormat {
         }
     }
 }
+
+/// Control when colored output is used.
+#[derive(Copy, Clone, Hash, Debug, PartialEq, Eq, PartialOrd, Ord, Default, clap::ValueEnum)]
+pub(crate) enum TerminalColor {
+    /// Automatically detect if color support is available on the terminal.
```

Comment:
This sounds a bit like we would actually detect if the terminal emulator supports colors. What's actually happening is just a `isatty` check (+ checks for `NO_COLOR` etc). So I tend to write something like
```suggestion
    /// Display colors if the output goes to an interactive terminal.
```